### PR TITLE
fix: universal link scanning

### PR DESCRIPTION
**Pull request by amytang0**
Fixes RNBW-####
Figma link (if any):

## What changed (plus any additional context for devs)
Current Universal Link scanning can be fragile if the order of the params is different than expected (i.e. if `uri` is not the first parameter)

## What to test
<!-- 
Scan a QR code encoding: `https://rnbwapp.com/wc?uri=wc%3Acedabb93-88bc-4df5-a2d5-9e87f471ecfc%401%3Fbridge%3Dhttps%253A%252F%252Fl.bridge.walletconnect.org%26key%3Dff3937eb054a2e9c23dfc41203a2e750a68a852472417d1c3859c14dd6264302` for Android and iOS

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/hooks/useScanner.ts [hunk 1]
```diff
@@ -165,15 +165,15 @@ export default function useScanner(enabled: boolean, onSuccess: () => unknown) {
...
-      if (
-        urlObj?.protocol === 'https:' &&
-        urlObj?.pathname?.split('/')?.[1] === 'wc'
-      ) {
-        // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
-        const { uri } = qs.parse(urlObj.query.substring(1));
-        onSuccess();
-        return handleScanWalletConnect(uri);
+      if (urlObj?.protocol === 'https:' && urlObj?.searchParams !== null) {
+        // eslint-disable-next-line no-unsafe-optional-chaining
```
Comment (amytang0) [inline src/hooks/useScanner.ts:169]:
```suggestion
        // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
```

File: src/hooks/useScanner.ts [hunk 2]
```diff
@@ -165,15 +165,15 @@ export default function useScanner(enabled: boolean, onSuccess: () => unknown) {
       }
       // Walletconnect via universal link
       const urlObj = new URL(data);
-      if (
-        urlObj?.protocol === 'https:' &&
-        urlObj?.pathname?.split('/')?.[1] === 'wc'
-      ) {
+      if (urlObj?.protocol === 'https:' && urlObj?.searchParams !== null) {
```
Comment (amytang0) [inline src/hooks/useScanner.ts:168]:
```suggestion
      if (urlObj?.protocol === 'https:' && 
      urlObj?.searchParams !== null) {
```

**Comment by jkadamczyk** [comment]
Created a Linear ticket to track work on this PR
https://linear.app/rainbow/issue/APP-12/external-pr-with-changes-to-universal-link-scanning

**Comment by skylarbarrera** [comment]
hey @amytang0 cherry picked and opened a new PR: https://github.com/rainbow-me/rainbow/pull/4368 so it'll run our CI and we can test everything out 

awesome catch, wanna make sure u get your OS credit 😎

**Pull request closed without merging**